File: bench/fp.py
import time
N = 20000000
t0 = time.perf_counter()
s = 0.0
sign = 1.0
i = 0
while i < N:
    s += sign / float(2 * i + 1)
    sign = -sign
    i += 1
ms = (time.perf_counter() - t0) * 1000.0
print("RESULT %f" % (s * 4.0))
print("TIME_MS %.3f" % ms)

File: bench/fp.rs
use std::time::Instant;
const N: i64 = 20000000;
fn main() {
    let t0 = Instant::now();
    let mut s: f64 = 0.0;
    let mut sign: f64 = 1.0;
    let mut i: i64 = 0;
    while i < N {
        s += sign / ((2 * i + 1) as f64);
        sign = -sign;
        i += 1;
    }
    let ms = t0.elapsed().as_secs_f64() * 1000.0;
    println!("RESULT {:.6}", s * 4.0);
    println!("TIME_MS {:.3}", ms);
}
